### DISCUSSION TITLE
How to upload video to Telegram servers to send it multiple times?

### DISCUSSION THREAD

**Discussion by plammens** [python-telegram-bot/python-telegram-bot#3130] [Q&A] [answered]
The docs for [`telegram.Bot.send_video`](https://docs.python-telegram-bot.org/en/stable/telegram.bot.html#telegram.Bot.send_video) say (emphasis mine):

> Video file to send. **Pass a file_id as String to send an video file that exists on the Telegram servers (recommended)**, pass an HTTP URL as a String for Telegram to get an video file from the Internet, or upload a new one using multipart/form-data. Lastly you can pass an existing [telegram.Video](https://docs.python-telegram-bot.org/en/stable/telegram.video.html#telegram.Video) object to send.

I want my bot to send the same video in response to a command. To save resources, ideally this video should be uploaded only once to Telegram, and then its file_id should be used to send the video in a chat, as mentioned above.

How do I upload a video to Telegram and get its file_id?
Similar question for a video note.

**Comment by harshil21** [comment, marked as answer]
See https://github.com/python-telegram-bot/python-telegram-bot/wiki/Code-snippets#working-with-files-and-media

just do:
```py
msg = update.message.send_video(...)
file_id = msg.video.file_id
```

**Comment by Poolitzer** [comment]
You can not proactively load files into Telegram. You need to pick a chat and upload to it (e.g. open a channel or a group with only the bot in it)
